twilly_cli/src/sync/maps.rs

```diff
@@ -80,6 +89,187 @@ pub async fn choose_map_action(twilio: &Client, sync_service: &SyncService) {
+    6. Copy all items from the temporary map into the new map
+
+We will not delete the temporary map after the process has completed.
+You can remove this using the CLI after you've confirmed the rename was successful.
+Would you like to continue?";
+                    let confirm_operation = Confirm::new(confirmation_message)
+                        .with_placeholder("N")
+                        .with_default(false);
+
+                    let confirmation_result = prompt_user(confirm_operation);
+
+                    if let None = confirmation_result {
+                        break;
+                    }
+
+                    println!("Starting map rename process");
+
+                    // create temporary map
+                    println!("(1/6) Creating temporary map");
+                    let temp_map_result = twilio
+                        .sync()
+                        .service(&sync_service.sid)
+                        .maps()
+                        .create(CreateMapParams {
+                            ttl: None,
+                            unique_name: Some(String::from(format!(
+                                "temp-{}",
+                                selected_sync_map.unique_name
+                            ))),
+                        })
+                        .await;
+
+                    if let Err(error) = temp_map_result {
+                        println!("Errored: Failed to create map: {:?}", error);
+                        break;
+                    }
+
+                    let temp_map = temp_map_result.unwrap();
+
+                    // clone all items into temp map
+                    println!("(2/6) Clone items into temporary map");
+                    let fetch_items_result = twilio
+                        .sync()
+                        .service(&sync_service.sid)
+                        .map(&selected_sync_map.sid)
+                        .mapitems()
+                        .list(ListParams {
+                            bounds: None,
+                            from: None,
+                            order: None,
+                        })
+                        .await;
+
+                    if let Err(error) = fetch_items_result {
+                        println!("Errored: Failed to fetch current map items: {:?}", error);
+                        break;
+                    }
+
+                    let items = fetch_items_result.unwrap();
+
+                    for item in items.iter() {
+                        let create_item_result = twilio
+                            .sync()
+                            .service(&sync_service.sid)
+                            .map(&temp_map.sid)
+                            .mapitems()
+                            .create(CreateMapItemParams {
+                                key: String::from(&item.key),
+                                data: &item.data,
+                                collection_ttl: None,
```

Comment:
Should the TTL parameters be preserved when renaming a map? Alternatively maybe worth stating that any set TTL's will be ignored.

Reply:
I'm unsure about this one.

At a map level I think it can result in unexpected behaviour from the map potentially deleting itself during the process.

This same thing could also happen to items, but I think it is technically possible there.